istioctl/pkg/waypoint/waypoint.go

```diff
@@ -310,6 +427,9 @@ func Cmd(ctx cli.Context) *cobra.Command {
 	}
 
 	waypointApplyCmd.PersistentFlags().StringVarP(&revision, "revision", "r", "", "The revision to label the waypoint with")
+	waypointApplyCmd.PersistentFlags().BoolVarP(&waitReady, "wait-ready", "W", false, "Wait for the waypoint to be ready and synced")
```

Comment:
I would just have one flag, `--wait`. Not sure we really need a timeout too?

Reply:
The initial idea was to make it configurable to be consistent with istioctl install.
@hanxiaop also pointed that in this case it should be enough to rely on a default hardcoded timeout value.

Istioctl install uses 5 minutes as the default timeout, which I presume is to accommodate for slow image pulling speeds. However, in our tests, we wait for the waypoint pod is ready within 15 seconds (likely because the proxy image has already been pulled before).

Currently, I have removed the timeout flag and replaced it with a hardcoded timeout of 90 seconds. Do you think we should align to the 5 minutes of timeout in "istioctl install" or opt to keep/lower the current value of 90 seconds?